File: twoSum/twoSum.go
func twoSum(nums []int, target int) []int {
    a := make(map[int]int)

    for i:=0; i<len(nums); i++ {
        b := target - nums[i]
        _, ok := a[b]
        if ok {
            x := []int{i, a[b]};
            return x;
        }

        a[nums[i]] = i;
    }

    return []int{};
}

File: twoSum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let a = {}
    let res = []

    nums.forEach((item, index) => {

        const b = target - item
        if (b in a) {
            res = [index, a[b]]
        }

        a[item] = index
    })

    return res;
};